lib/karafka/connection/consumer.rb

```diff
@@ -3,22 +3,21 @@ module Connection
     # Class that consumes messages for which we listen
     class Consumer
       # Consumes a message (does something with it)
-      # It will route a message to a proper controller and execute it
-      # Logic in this method will be executed for each incoming message
+      # It will execute a scheduling task from a proper controller based on a message topic
       # @note This should be looped to obtain a constant listening
       # @note We catch all the errors here, to make sure that none failures
       #   for a given consumption will affect other consumed messages
       #   If we would't catch it, it would propagate up until killing the Celluloid actor
-      # @param controller_class [Karafka::BaseController] base controller descendant class
       # @param message [Poseidon::FetchedMessage] message that was fetched by poseidon
-      def consume(controller_class, message)
-        controller = Karafka::Routing::Router.new(
-          Message.new(controller_class.topic, message.value)
-        ).build
+      def consume(message)
+        controller = Karafka::Routing::Router.new(message.topic).build
+        # We wrap it around with our internal message format, so we don't pass around
+        # a raw Poseidon message
+        controller.params = Message.new(message.topic, message.value)
```

Comment:
Seems, it is only one place where we use Message wrapper, isn't it?
I think it's enough to send `message` as a param and change `Message` class implementation.